### PR TITLE
Ui and balance changes

### PR DESCRIPTION
**Pull request by Flam3sDOTA**
Added Health Bars of the Kings
Added Information Button on Top Left Corner
Added Information Panel
Updated Loading Screen
Updated Resources Panel
Removed Overthrow Game Information Panel that Had Contributors Panel

Undead Builder:
Zombie:
Decreased Armor from 3 to 0

Mutilated Butcher:
Decreased Armor from 3 to 1

Greater Zombie:
Upgrade Cost Increased from 65 to 75
Decreased Health from 325 to 290
Decreased Armor from 3 to 1

Netherdrake:
Decreased Armor from 2 to 0
Increased Attack Rate from 0.8 to 1(Slower Attack Speed)
Decreased Health from 350 to 300
Decreased Attack Damage from 120 to 100

### CODE REVIEW

**Comment by Flam3sDOTA** [comment]
New Resources Panel:
NewPanel

New Loading Screen:
Loading_Screen


New Information Button:
InformationButton

New Information Panel(Hero Avatars are WEBMs not Images.):
InformationPanel

**Comment by Flam3sDOTA** [comment]
Healthbars:
Healthbars

**Comment by n-gao** [comment]
I will check that in a moment looks like a big update. Looks great!

File: content/dota_addons/legion_td/panorama/layout/custom_game/custom_loading_screen.xml [hunk 1]
```diff
@@ -29,4 +29,4 @@
             </Panel>
         </Panel>
     </Panel>
-</root>
```
Comment (n-gao) [inline content/dota_addons/legion_td/panorama/layout/custom_game/custom_loading_screen.xml]: As convention let files end with an empty new line

File: content/dota_addons/legion_td/panorama/layout/custom_game/custom_ui_manifest.xml [hunk 2]
```diff
@@ -43,16 +43,21 @@
...
     var newUI = $.GetContextPanel().GetParent().GetParent().FindChildTraverse("HUDElements").FindChildTraverse("lower_hud").FindChildTraverse("center_with_stats").FindChildTraverse("center_block");
-
-    newUI.FindChildTraverse("StatBranch").style.visibility = "collapse";
-
+	var NoGlyphAndRadar = $.GetContextPanel().GetParent().GetParent().FindChildTraverse("HUDElements").FindChildTraverse("minimap_container");
+	var NoTPScrollUI = $.GetContextPanel().GetParent().GetParent().FindChildTraverse("HUDElements").FindChildTraverse("lower_hud").FindChildTraverse("center_with_stats").FindChildTraverse("center_block");
+	var NoKDA = $.GetContextPanel().GetParent().GetParent().FindChildTraverse("HUDElements").FindChildTraverse("stackable_side_panels");
+    
+	newUI.FindChildTraverse("StatBranch").style.visibility = "collapse";
+	NoGlyphAndRadar.FindChildTraverse("GlyphScanContainer").style.visibility = "collapse";
```
Comment (n-gao) [inline content/dota_addons/legion_td/panorama/layout/custom_game/custom_ui_manifest.xml]: Could we get the minimap container back? It looks weird without even though glyph and radar are irrelevant for this game mode. The best option would probably to have a custom container.

**Comment by n-gao** [comment]
Could you add an image where I can see where the healthbars are located?

File: content/dota_addons/legion_td/panorama/layout/custom_game/team_select_team.xml [hunk 3]
```diff
@@ -1,24 +1,22 @@
...
+			<Panel id="TeamHeader">
+				<Panel id="TeamLogo" />
+				<Label id="TeamNameLabel" text="TEAM_NAME" />
+				<Label id="JoinTeamLabel" text="#custom_game_team_select_join" />
+			</Panel>
+			<Panel id="TeamPlayerDivider" />
+			<Panel id="PlayerList" />
+		</Panel>
+	</Button>
+</root>
```
Comment (n-gao) [inline content/dota_addons/legion_td/panorama/layout/custom_game/team_select_team.xml]: Could you make this to fill the whole container? Atm, there is a margin at the top and bottom that would look better if removed.
image

Comment (n-gao) [reply content/dota_addons/legion_td/panorama/layout/custom_game/team_select_team.xml]: Also please have an empty line at the end of each file

File: content/dota_addons/legion_td/panorama/scripts/custom_game/team_select.js [hunk 4]
```diff
@@ -327,4 +326,4 @@ function UpdateTimer() {
 
     // Register a listener for the event which is broadcast whenever a player attempts to pick a team
     $.RegisterForUnhandledEvent("DOTAGame_PlayerSelectedCustomTeam", OnPlayerSelectedTeam);
-})();
+})();
```
Comment (n-gao) [inline content/dota_addons/legion_td/panorama/scripts/custom_game/team_select.js]: Please have and empty line at the end

File: content/dota_addons/legion_td/panorama/styles/custom_game/custom_loading_screen.css [hunk 5]
```diff
@@ -1,15 +1,17 @@
...
 .OverthrowGameInfo {
     width: 100%;
     height: 100%;
     background-image: url("file://{images}/custom_game/game_info/loading.png");
+	background-size: 100% 100%;
 }
 
 .OverthrowGameInfo2 {
-    height: 803px;
+	height: 803px;
```
Comment (n-gao) [inline content/dota_addons/legion_td/panorama/styles/custom_game/custom_loading_screen.css:14]: this is wrongly indented

File: content/dota_addons/legion_td/panorama/styles/custom_game/custom_loading_screen.css [hunk 6]
```diff
@@ -1,15 +1,17 @@
 @define TotalLength: 30s;
 @define TimingFunc: linear;
 @define IterationCount: 1;
+@define AnimationDelay: 0s;
 
 .OverthrowGameInfo {
     width: 100%;
     height: 100%;
     background-image: url("file://{images}/custom_game/game_info/loading.png");
+	background-size: 100% 100%;
```
Comment (n-gao) [inline content/dota_addons/legion_td/panorama/styles/custom_game/custom_loading_screen.css:10]: this is wrongly indented

File: content/dota_addons/legion_td/panorama/styles/custom_game/custom_loading_screen.css [hunk 7]
```diff
@@ -47,11 +49,13 @@ Label {
...
     font-size: 70px;
-    font-weight: bold;
     margin-top: 32px;
     margin-left: 20px;
-    color: white;
     text-transform: uppercase;
+	
+	color: #fff;
+	font-weight: semi-bold;
+	text-shadow: 0px 0px 4px #CD6BFB;
```
Comment (n-gao) [inline content/dota_addons/legion_td/panorama/styles/custom_game/custom_loading_screen.css:58]: this is wrongly indented

File: content/dota_addons/legion_td/panorama/styles/custom_game/custom_loading_screen.css [hunk 8]
```diff
@@ -62,17 +66,18 @@ Label {
...
 .Header {
     font-size: 35px;
-    font-weight: bold;
     margin-top: 36px;
     text-transform: uppercase;
-    color: yellow;
+	
+	color: #fff;
+	font-weight: semi-bold;
+	text-shadow: 0px 0px 4px #CD6BFB;
```
Comment (n-gao) [inline content/dota_addons/legion_td/panorama/styles/custom_game/custom_loading_screen.css:74]: this is wrongly indented

File: content/dota_addons/legion_td/panorama/styles/custom_game/custom_loading_screen.css [hunk 9]
```diff
@@ -185,3 +189,48 @@ Label .xp {
...
+.GameInformationPanel
+{
+	background-image: url("file://{images}/custom_game/game_info/topbar.png");
+	background-repeat: no-repeat;
+	margin-top: 120px;
+	margin-left: 25px;
+	border-radius: 10px;
+	width: 690px;
+	height: 164px;
+	opacity: 1.5;
```
Comment (n-gao) [inline content/dota_addons/legion_td/panorama/styles/custom_game/custom_loading_screen.css:202]: wrong indention level

File: content/dota_addons/legion_td/panorama/styles/custom_game/custom_loading_screen.css [hunk 10]
```diff
@@ -185,3 +189,48 @@ Label .xp {
...
+	height: 670px;
+	background-color: #000000BA;
+	margin-left: 55px;
+	margin-top: 100px;
+	border-style: solid;
+    border-width: 1px;
+    border-color: #444444;
+	border-radius: 7px;
+  visibility: collapse;
+	flow-children: down;
```
Comment (n-gao) [inline content/dota_addons/legion_td/panorama/styles/custom_game/custom_loading_screen.css]: inconsistently indented

File: content/dota_addons/legion_td/panorama/styles/custom_game/custom_loading_screen.css [hunk 11]
```diff
@@ -185,3 +189,48 @@ Label .xp {
...
+.InfoLabel
+{
+	margin-top: 15px;
+	margin-left: 15px;
+	horizontal-align: center;
+	font-size: 55px;
+	color: orangered;
+	text-shadow: 2px 2px 2px #000000;
+	font-weight: bold;
+	text-transform: uppercase;
```
Comment (n-gao) [inline content/dota_addons/legion_td/panorama/styles/custom_game/custom_loading_screen.css:234]: should be lower indented
Please keep one indention level

File: content/dota_addons/legion_td/panorama/styles/custom_game/information_panel.css [hunk 12]
```diff
@@ -0,0 +1,141 @@
...
+	overflow: clip;
+	flow-children: none;
+	border: 1px solid #5e6869;	
+	align: center center;
+	margin-bottom: 100px;
+	
+	background-color: #000;
+	background-image: url("s2r://panorama/images/backgrounds/gallery_background_png.vtex");
+	background-size:100% 100%;
+    box-shadow: fill #00000066 -4px -4px 8px 8px;
```
Comment (n-gao) [inline content/dota_addons/legion_td/panorama/styles/custom_game/information_panel.css]: keep one indention level (which should be the one I used before)

Comment (n-gao) [reply content/dota_addons/legion_td/panorama/styles/custom_game/information_panel.css]: I will stop marking this for the rest of the request but you should still change it. Thanks!

**Review by n-gao — CHANGES_REQUESTED** [review]
This is great! Thanks a lot! Please look at my comments and incorporate some changes but despite from that great work!

File: content/dota_addons/legion_td/panorama/layout/custom_game/custom_ui_manifest.xml (hunk 2 above)
Comment (Flam3sDOTA) [reply content/dota_addons/legion_td/panorama/layout/custom_game/custom_ui_manifest.xml]: I will try it in the next update, Kinda busy in the next days, Would you like me to update the game right after you merge or the container is a big problem and you wanna wait before updating?

**Comment by Flam3sDOTA** [comment]
> Could you add an image where I can see where the healthbars are located?

Healthbars

It is this high because there is option to use "bigger minimap" and I don't want to conflit with it.

File: content/dota_addons/legion_td/panorama/layout/custom_game/custom_ui_manifest.xml (hunk 2 above)
Comment (n-gao) [reply content/dota_addons/legion_td/panorama/layout/custom_game/custom_ui_manifest.xml]: Just get the old container back before merging

**Comment by n-gao** [comment]
> > Could you add an image where I can see where the healthbars are located?
> 
> Healthbars
> 
> It is this high because there is option to use "bigger minimap" and I don't want to conflit with it.

Thanks!

**Comment by n-gao** [comment]
Please adjust the last few code snippets and give back the old frame (you can, later on, put a new one in)
Also, you can make larger commits with multiple files. No need to commit every file as a single commit

**Comment by Flam3sDOTA** [comment]
> Please adjust the last few code snippets and give back the old frame (you can, later on, put a new one in)
> Also, you can make larger commits with multiple files. No need to commit every file as a single commit

Yeah I know, I went 1 by 1 conversation, didin't realize they were same files

File: content/dota_addons/legion_td/panorama/layout/custom_game/custom_ui_manifest.xml (hunk 2 above)
Comment (Flam3sDOTA) [reply content/dota_addons/legion_td/panorama/layout/custom_game/custom_ui_manifest.xml]: I don't like the the old container, and I will make new one merging it with king's healths and will expand that panel with attack-health-regen modifiers etc so can it stay like this until update? not a big deal.

**Comment by Flam3sDOTA** [comment]
Is there anything needs to be done? (apart from container since I don't have time for 2 days)

**Comment by n-gao** [comment]
Correct the last few indentions and reenable the old container. Just disable the container when the new GUI is done and we are good 👍

**Comment by Flam3sDOTA** [comment]
> Correct the last few indentions and reenable the old container. Just disable the container when the new GUI is done and we are good 👍

Oh you mean revert the "container disable on custom_ui_mainfest" ? I did it but the map overlay didn't come back, so it is not related to that. I will just do a new map overlay in 2 day or so, can we update it before I sleep? It is kinda 1 am here :P

**Comment by n-gao** [comment]
Why did the map overlay not came back? Please get it back and there are still some incorrect indentions.

**Comment by n-gao** [comment]
For future reference, use exactly 4 spaces for indention don't use tabs the correct style is in [.editorconfig](https://github.com/n-gao/LegionTD-Reborn/blob/master/.editorconfig). Make sure your editor loads that file correctly. If you use vscode (which I recommend) install the editorconfig extension.

**Comment by n-gao** [comment]
I reenabled to old minimap UI. This is good now.

**Comment by Flam3sDOTA** [comment]
Ah alright then, I don't know why minimap UI was gone, sometimes tools gets buggy. Yeah I will use the vscode.

**Comment by Flam3sDOTA** [comment]
Ah alright then, I don't know why minimap UI was gone, sometimes tools gets buggy. Yeah I will use the vscode.

Let me know if anything else is needed.

**Pull request merged**